lib/eofs/standard.py

```diff
@@ -152,7 +158,18 @@ def __init__(self, dataset, weights=None, center=True, ddof=1):
         dataNoMissing = self._data[:, nonMissingIndex]
         # Compute the singular value decomposition of the design matrix.
         try:
-            A, Lh, E = np.linalg.svd(dataNoMissing, full_matrices=False)
+            if has_dask and isinstance(dataNoMissing, dask.array.Array):
+                # Use the parallel Dask algorithm
+                dsvd = dask.array.linalg.svd(dataNoMissing)
```

Comment:
It looks like this doesn't raise an error (or the same error as numpy) when the array contains all missing values. This is the reason for the "did not raise" test failures. It may be that the test needs updating as it relies on numpy behaviour to ensure the input contains only missing values.

Reply:
There appears to have been a change in numpy behaviour. `test_input_with_all_missing_values` fails with the most recent version of numpy, but succeeds if I downgrade numpy to 1.13.3 (the version used in the most recent successful 'master' test)

It appears that support for SVD on empty arrays was added by  https://github.com/numpy/numpy/issues/8654

Probably the simplest thing to do would be to add an explicit check for an empty `dataNoMissing` array and raise a ValueError.